src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,34 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<Machine> type) {
+        List<Machine> machines = new ArrayList<>();
+        machines.add(new Bulldozer("Caterpillar D9", 400, 3.5));
+        machines.add(new Excavator("Komatsu PC200", 150, 2.0));
+        machines.add(new Truck("Volvo FH", 500, 12.0));
```

Comment:
The `getAll` method does not use the `type` parameter to determine which specific subclass of `Machine` to instantiate. Instead, it currently adds instances of `Bulldozer`, `Excavator`, and `Truck` directly. Consider using the `type` parameter to dynamically create instances of the specified machine type.